src/git.rs

```diff
@@ -176,21 +178,33 @@ pub fn setup_update_branch(
         BranchType::Remote,
     );
 
+    let default_branch = repo
+        .find_branch(
+            &format!("origin/{}", &settings.default_branch),
+            BranchType::Remote,
+        )
+        .map_err(SetupUpdateBranchError::FindDefaultBranch)?;
+
     let branch = if let Ok(b) = update_branch {
         let update_branch_commit = b
             .get()
             .peel_to_commit()
             .map_err(SetupUpdateBranchError::PeelUpdateBranchCommit)?;
-        if update_branch_commit.author().email() != Some(&settings.author.email) {
+        let default_branch_commit = b
```

Comment:
Shouldn't it be `default_branch.get().peel_to_commit()`? :thinking:

Reply:
AFAIU, currently `update_branch_commit.id() == default_branch_commit.id()` is always true

Reply:
Okay, sorry, *pushing* a fixup